### ISSUE TITLE
Bug: Constructor invocations in platform code do not get correct default values

### ISSUE THREAD

**Issue by eernstg** [dart-lang/reflectable#180]
With version 2.1.0 of reflectable the dependencies have been updated to allow for analyzer 0.37.0 (and many other dependencies have been updated as well). One of the consequences of doing this is that reflectable cannot get information from the analyzer about source code constructs (`AstNode`) for declarations in platform libraries, but only for those declarations.

Platform libraries are the ones which are imported using `'dart:N'` for some name `N` (e.g., `'dart:html'` or `'dart:math'`, except `'dart:ui'` which is not a platform library).

This means that reflectable cannot generate code for supporting constructor invocations with the correct default values, for constructors declared in a platform library. Instead, every optional parameter is given the default value `null`. This means that constructor invocation via a class mirror will not faithfully mimic a regular (non-reflective) constructor invocation when optional parameters with a default value different from null are omitted.

This issue is blocked on getting a bit more flexibility from the build package (in particular, https://github.com/dart-lang/build/issues/733 is relevant), which could allow reflectable to request a resolver which uses full libraries rather than summaries for the SDK.

**Comment by fertrig** [comment]
Could you please confirm that this a new issue present in 2.1.0? I think this functionality was working on previous versions, like 2.0.12.

**Comment by eernstg** [comment]
Yes, it was working, it is a new issue: The analyzer used by the build package uses summaries rather than full libraries for platform libraries, so there is no information about metadata nor about default values of parameters.
